### ISSUE TITLE
[Feature Request?] Sync listening progress to Jellyfin/Emby

### ISSUE THREAD

**Issue by rthomasv3** [LeoKlaus/plappa#176]
**Describe the bug**
When the app is linked to a jellyfin server, listening progress for a book is not synced back to the server.

**To Reproduce**
1. Connect to a jellyfin server
2. Listen to any book for a period of time
3. Go to the jellyfin server through the web interface or jellyfin app and the progress has not been updated.

**Expected behavior**
When listening to an audiobook I would expect the listening progress to be synced back to the server so I can continue listening from where I left off on another device.

**Environment**
 - Device: iPhone 13
 - OS: iOS 17.5.1
 - Version: 1.4.0
 - Server Version: Jellyfin 10.9.7

**Comment by LeoKlaus** [comment]
Hey Robert, thank you for reaching out!

At the moment, this is intentional. I've found the progress tracking in Jellyfin to be rather lacking (especially for audiobooks made up of multiple files). As of now, all progress tracking is performed locally and synced via iCloud when using Jellyfin or Emby.

**Comment by rthomasv3** [comment]
Yeah, makes sense, I assumed there was a pretty good reason. Would it be possible to get it added as an optional/experimental feature? If not, it looks like audio book shelf will mostly fit my needs and I can probably just switch everything over.

**Comment by LeoKlaus** [comment]
> Would it be possible to get it added as an optional/experimental feature?

Absolutely, I think I even implemented this half way a while ago but never enabled it. The big issue with this is choosing a source of truth.

I highly suggest testing ABS, the setup is very simple and it's fairly light on resources. It also does a much better job finding metadata for audiobooks compared to Jellyfin.

**Comment by zhimiaoli** [comment]
I also need this feature, using both Emby and ABS , but ABS has some bugs with the Podcast library,Emby is better at handling books if I add audio files to a book periodicaly.

**Comment by moeinio** [comment]
Since progress is already synced through iCloud, is it possible to sync progress across different clients? Eg. From iPhone to iPad app? If ABS is hosted locally, you'd need to be connected to it to sync listening progress. That is literally the only reason why I still have to use apple books instead of this app.

**Comment by LeoKlaus** [comment]
> Since progress is already synced through iCloud, is it possible to sync progress across different clients? Eg. From iPhone to iPad app? If ABS is hosted locally, you'd need to be connected to it to sync listening progress. That is literally the only reason why I still have to use apple books instead of this app.

Progress is only synced through iCloud with Jellyfin/Emby (in that case it is of course synced across all clients). When using ABS, your server is used exclusively for progress syncing. I could add a toggle to use iCloud instead of ABS for syncing, though.

**Comment by moeinio** [comment]
That would be amazing. I use the download feature on multiple devices to be able to listen while not being connected to ABS, it would be nice if the clients could sync progress on app start.

**Comment by jvimedia** [comment]
For my use case, this is unfortunately, as it stands, a deal breaker for plappa, as I rely heavily on Jellyfin’s web interface, and users in my environment also use the Jellyfin TV app.

Syncing progress is imho essential. A common approach in other jellyfin  apps is to compare the last update timestamp and trust the most recent value. This feels intuitive and aligns with user expectations from my experience. If syncing fails, users typically scrub to their last known position manually—similar to how Audible handles offline playback—but at least it’s a best-effort fallback. 

I am happy to test or contribute.

**Comment by pope7** [comment]
This is becoming a bit of (and the only) deal-breaker for me as well! My use case is that I listen using plappa in the car, and my desktop at work. 

I don't need any sort of 'smart' sync, but I'd LOVE a '...'-button context menu option to 'Push Playback to Jellyfin' and 'Pull Playback from Jellyfin'. A quick manual action to save me the trauma (hah) of having to use jellyfin's garbage seekbar every day.

Adore this app. You've done incredible work.

**Comment by LeoKlaus** [comment]
I've just released a first TestFlight beta for the upcoming 2.0 release of plappa which does include some sync to Jellyfin. It only works with single file items that Jellyfin treats as audiobooks. For all other items, only local progress tracking is supported in that configuration. You can enable server sync during the onboarding process.


I'll just copy/paste the post I made about the beta in Discord as there are some caveats:

It still has a lot of placeholder UI in it, but most of the core functionality should be there now.

For now, I'm specifically looking for feedback on reliability and stability.

**Important warning:** There are no migrations from the old version of plappa in place yet and the new beta will replace the old app. To prevent dead files clogging up your phone, please manually uninstall plappa before installing the beta. I**f you use plappa with Jellyfin, this means potentially losing playback progress and session history!**

Here's the TestFlight link (this won't be published to the existing groups for now):
https://testflight.apple.com/join/aup9keaD
